### DISCUSSION TITLE
ac power binary sensor

### DISCUSSION THREAD

**Discussion by jn3va** [Dilbert66/esphome-vistaECP#65] [Q&A] [answered]
My alarm panel is on ac power and all the LEDs inside the panel look good,  but vistaECP keeps reporting that ac power is off/unplugged.  Anyone else seeing this?  I can reboot the vistaECP or manually reset the binary sensor using Developer Tools and home assistant will show it as "Plugged in", but eventually it will reset to indicate "unplugged" sometime within 24 hours, but I can't seem to catch anything in the ESPHome logging to see when or what message is sent..   

The documentation looks like its problem off - but home assistant is interpreting "off" as "unplugged" and plugged in when its "on".

Does anyone know any other tests to run at the alarm keypad to get more info on what the panel sees as the power state, or issues?

**Comment by Dilbert66** [comment]
Are you using the latest version of the library? There was an issue with incorrect AC reporting with one of the earlier versions I had up a while ago.

**Reply by jn3va** [reply]
I need to update - Looks like the file time stamps are may 2021. I'll pull down the latest this week or next weekend and re-deploy.  
Thanks!
Joe

**Reply by jn3va** [reply]
I did a quick replace of the vistaEcpInterface folder and recompiled.  The log output in ESP Home looks like it reading sensors fine, and the AC sensor looks good now; but I'm not getting anything in the line1 / line 2 entities.  The Alarm panel card shows "unknown"  I reverted back to the working version and will check this again and look at it closer when I have more time later this week.

**Comment by Dilbert66** [comment]
Compare the yaml file as far as the line1/lin2 sensors.  I can't remember if anything else changed to affect the line1/lin2 displays.

**Reply by jn3va** [reply]
In the new YAML for EPS_Home this is the only change I found

 ```
 ##esp32
  #rxPin: "22" #GPIO pin to use for data receive (yellow line)
  #txPin: "21" #GPIO pin to use for data transmit (green line)
  #monitorPin: "18" #GPIO pin to use for monitoring module traffic such as RF or Expanders . Set to -1 to disable
```

I'm using an ESP32.  On the version that works it looks like you were originally using GPIOs as follows:

 ```
#define D1 18
#define D2 19
#define D5 21
```

Do you recall in the previous version (from around May 2021) 
Is D1=rx, D2=tx and D5=monitor?

**Reply by jn3va** [reply]
Looks like the define statements in the vistaAlarm.h were taking precedent over the #defines in the yaml file.  I manually changed the vistaAlarm.h as follows and its working now!   

//esp32 use gpio pins 4,13,16-39 
//#define RX_PIN 22
//#define TX_PIN 21
//#define MONITOR_PIN 18 // pin used to monitor the green TX line (3.3 level dropped from 12 volts
#define RX_PIN 18
#define TX_PIN 19
#define MONITOR_PIN 21 // pin used to monitor the green TX line (3.3 level dropped from 12 volts

**Comment by Dilbert66** [comment, marked as answer]
I suspect you had not uncommented the YAML values. ie remove the # sign in front. . In the YAML a # sign is a comment which differs to c++ where # is a compiler directive.
